src/AppInstallerRepositoryCore/PackageDependenciesValidation.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include <AppInstallerVersions.h>
+#include <winget/Manifest.h>
+#include <winget/ManifestValidation.h>
+#include <SQLiteWrapper.h>
+#include <PackageDependenciesValidation.h>
+#include <Microsoft/Schema/1_4/DependenciesTable.h>
+#include "Microsoft/Schema/1_0/ManifestTable.h"
+#include <winget/DependenciesGraph.h>
+
+namespace AppInstaller::Repository
+{
+    using namespace Microsoft::Schema::V1_4;
+    using namespace Microsoft::Schema::V1_0;
+
+    namespace
+    {
+        std::vector<AppInstaller::Manifest::Dependency> GetDependencies(
+            const Manifest::Manifest& manifest, AppInstaller::Manifest::DependencyType dependencyType)
+        {
+            std::vector<AppInstaller::Manifest::Dependency>  dependencies;
+
+            for (const auto& installer : manifest.Installers)
+            {
+                installer.Dependencies.ApplyToAll([&](AppInstaller::Manifest::Dependency dependency)
+                    {
+                        if (dependency.Type == dependencyType)
+                        {
+                            dependencies.emplace_back(dependency);
+                        }
+                    });
+            }
+
+            return dependencies;
+        }
+
+        std::optional<std::pair<SQLite::rowid_t, Utility::Version>> GetPackageLatestVersion(
+            SQLiteIndex* index, Manifest::string_t packageId, std::set<Utility::Version> exclusions = {})
+        {
+            SearchRequest request;
+            request.Filters.emplace_back(PackageMatchField::Id, MatchType::Exact, packageId);
+            std::optional<std::pair<SQLite::rowid_t, Utility::Version>> result;
+
+            auto results = index->Search(request);
+
+            if (results.Matches.empty())
+            {
+                return result;
+            }
+
+            auto packageRowId = results.Matches[0].first;
+
+            auto vac = index->GetVersionKeysById(packageRowId);
+
+            Utility::Version maxVersion = Utility::Version::CreateUnknown();
+
+            for (auto& v : vac)
+            {
+                auto currentVersion = v.GetVersion();
+                if (exclusions.find(currentVersion) != exclusions.end())
+                {
+                    continue;
+                }
+
+                if (maxVersion.IsUnknown() || currentVersion > maxVersion)
```

Comment:
>if (maxVersion.IsUnknown() || currentVersion > maxVersion) [](http://example.com/codeflow?start=16&length=58)

1. Unkown always compares as lower.
2. The version come back in sorted order, highest first, already.